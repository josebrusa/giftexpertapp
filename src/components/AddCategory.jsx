import { PropTypes } from "prop-types";
import { useState } from "react"


export const AddCategory = ( { onNewCategory } ) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = ( { target } ) => {
        setInputValue( target.value )
    }

    const onSubmit = ( event ) => {
        event.preventDefault();
            if( inputValue.trim().length <= 1) return;
            setInputValue('');
            onNewCategory( inputValue.trim() )
    }

    return (
        <form onSubmit={ onSubmit } >
            <input
                placeholder="busca tu gifs"
                type="text"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}

