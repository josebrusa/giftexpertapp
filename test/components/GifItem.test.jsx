import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"


describe('Pruebas en <GifItem />', () => {
    
    const title = 'Goku';
    const url = 'https://media0.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=b4594af3wkjgme17ipph4m951gl81jxch43390iaao8g2v3e&ep=v1_gifs_search&rid=giphy.gif&ct=g';


    test('debe realizar match con el snapshot ', () => {

        const {container } = render( <GifItem title={title} url={url} />);
        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el url y el alt indicado', () => {
        
        render( <GifItem title={ title } url={ url } />);
        // screen.debug();

        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('debe de mostrar el titulo en el componenete', () => {

        render( <GifItem url={ url } title={ title } />);

        // screen.debug();
        expect( screen.getByText( title ) ).toBeTruthy();
    });
    
});
