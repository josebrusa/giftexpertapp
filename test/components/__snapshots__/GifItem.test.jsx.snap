// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe realizar match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://media0.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=b4594af3wkjgme17ipph4m951gl81jxch43390iaao8g2v3e&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
